// @directus/shared/dist/cjs/utils/is-extension.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isExtensionObject = exports.isExtensionPackage = exports.isHybridExtension = exports.isApiExtension = exports.isAppExtension = exports.isExtension = void 0;
const constants_1 = require("../constants");
function isExtension(type) {
    return constants_1.EXTENSION_TYPES.includes(type);
}
exports.isExtension = isExtension;
function isAppExtension(type) {
    return constants_1.APP_EXTENSION_TYPES.includes(type);
}
exports.isAppExtension = isAppExtension;
function isApiExtension(type) {
    return constants_1.API_EXTENSION_TYPES.includes(type);
}
exports.isApiExtension = isApiExtension;
function isHybridExtension(type) {
    return constants_1.HYBRID_EXTENSION_TYPES.includes(type);
}
exports.isHybridExtension = isHybridExtension;
function isExtensionPackage(type) {
    return constants_1.EXTENSION_PACKAGE_TYPES.includes(type);
}
exports.isExtensionPackage = isExtensionPackage;
function isExtensionObject(obj, arr) {
    if (!obj.type)
        return false;
    return arr.includes(obj.type);
}
exports.isExtensionObject = isExtensionObject;
